qiita_plugins/target_gene/tgp/util.py

```diff
@@ -0,0 +1,194 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+import time
+import requests
+import threading
+from json import dumps
+from subprocess import Popen, PIPE
+
+JOB_COMPLETED = False
```

Comment:
This sort of thing would be really hard to unit-test (since it's global).

Would it be worthwhile to open up an issue to gather ideas on how to test out the heartbeat functionality?